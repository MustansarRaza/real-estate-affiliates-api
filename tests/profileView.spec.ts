import { loadSequelizeFixtures, useRequestFactory } from "./support";
import { ModelFixtureNames } from "./support/data";

describe("/profile", () => {
  beforeEach(async () => {
    await loadSequelizeFixtures(
      ModelFixtureNames.AGENTS,
      ModelFixtureNames.USERS,
      ModelFixtureNames.CONTACTS,
      ModelFixtureNames.AFFILIATES,
      ModelFixtureNames.AFFILIATE_CONTACTS,
      ModelFixtureNames.POTENTIAL_AFFILIATE_VERIFICATION_STATUS,
      ModelFixtureNames.POTENTIAL_AFFILIATES
    );
  });

  it("denies access to unauthenticated requests", async () => {
    const { body, header } = await useRequestFactory()
      .get("/profile")
      .expect(403);

    expect(body).toMatchSnapshot();
    expect(header).toMatchSnapshot();
  });

  it("returns the profile", async () => {
    const { body, header } = await useRequestFactory()
      .withVerifiedUser()
      .get("/profile")
      .expect(200);

    expect(body).toMatchSnapshot();
    expect(header).toMatchSnapshot();
  });

  it("returns 404 if no profile could be found for the user", async () => {
    // id of a valid keycloak user, but no profile in the db
    const ssoID = "caf59266-3b27-4e66-a47b-d99a42458b6e";

    const { body } = await useRequestFactory()
      .createUser({ id: ssoID })
      .get("/profile")
      .expect(404);

    expect(body).toMatchSnapshot();
  });
});
