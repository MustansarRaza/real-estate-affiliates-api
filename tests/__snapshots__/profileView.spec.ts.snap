// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/profile denies access to unauthenticated requests 1`] = `Object {}`;

exports[`/profile denies access to unauthenticated requests 2`] = `
Object {
  "cache-control": "private, no-store",
  "connection": "close",
  "strict-transport-security": "max-age=15552000; includeSubDomains",
  "transfer-encoding": "chunked",
  "x-content-type-options": "nosniff",
  "x-dns-prefetch-control": "off",
  "x-download-options": "noopen",
  "x-frame-options": "SAMEORIGIN",
  "x-xss-protection": "1; mode=block",
}
`;

exports[`/profile returns 404 if no profile could be found for the user 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Current user is not an affiliate",
      "name": "NotFound",
    },
  ],
}
`;

exports[`/profile returns the profile 1`] = `
Object {
  "address": "Disney Street 123",
  "affiliateType": "agency",
  "businessName": "Disney",
  "email": "walt@disney.com",
  "familyName": "Disney",
  "givenName": "Walt",
  "govIDNumber": "44443333",
  "govIDType": "NIC",
  "landLinePhoneNumbers": Array [],
  "locationID": 3,
  "mobilePhoneNumbers": Array [
    "+923344416658",
  ],
  "taxNumber": "128937934hsdhjdskf",
  "termsAndConditionsVersion": null,
  "username": "walt@disney.com",
}
`;

exports[`/profile returns the profile 2`] = `
Object {
  "cache-control": "private, no-store",
  "connection": "close",
  "content-length": "365",
  "content-type": "application/json; charset=utf-8",
  "etag": "W/\\"16d-CWeSuDD/XF30WLvcYu/1dc7UoVM\\"",
  "strict-transport-security": "max-age=15552000; includeSubDomains",
  "vary": "Accept-Encoding",
  "x-content-type-options": "nosniff",
  "x-dns-prefetch-control": "off",
  "x-download-options": "noopen",
  "x-frame-options": "SAMEORIGIN",
  "x-xss-protection": "1; mode=block",
}
`;
